CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,183 @@
+   * Requires the amount of stable token being bought/sold to be within the range `[minStableExchangeAmount, maxStableExchangeAmount]`.
+   * Deposits the full amount of the asset being sold into the contract.
+   * Records:
+     * The exchange as in the Proposed state.
+     * A struct with the following info is stored in a mapping with an `id` key:
+       * Which asset is being sold.
+       * The amount of the asset being sold.
+   * Returns:
+     * The `id` of the struct in the mapping.
+2. **`function approveExchangeProposal(uint256 id) external`** - Approves a proposed exchange.
+   * Only callable by the approver address.
+   * Marks the proposed exchange with the given `id` as approved.
+   * Records:
+     * The exchange as in the Approved state.
+     * The timestamp at which the approval occurred (ie `block.timestamp`).
+3. **`function cancelExchangeProposal(uint256 id) external`** - Cancels a proposed exchange.
+   * The permitted caller depends upon the state of the exchange proposal:
+     * If in the Proposed state, can only be called by the proposer.
+     * If in the Approved state, can only be called by Governance.
+     * If in any other state, cannot be called.
+   * Refunds the proposed exchange's deposited sell asset to the proposer.
+   * Records:
+     * The exchange as in the Cancelled state.
+4. **`function executeExchange(uint256 id) external`** - Executes an exchange.
+   * Callable by anyone.
+   * Requires the exchange proposal to have been approved.
+   * Requires the required waiting period for an approved exchange to have elapsed since the time it was approved (ie `block.timestamp - approval timestamp >= exchangeWaitPeriod`)
+   * Makes the exchange:
+     * If selling CELO and purchasing stable token:
+       * The CELO is sent to the Reserve.
+       * Stable token is minted to the exchange proposer according to the rate originally recorded when the exchange was proposed.
+     * If selling stable token and purchasing CELO:
+       * The stable token is burned.
+       * CELO is transferred from the Reserve to the exchange proposer according to the rate originally recorded when the exchange was proposed.
+   * Records:
+     * The exchange as in the Executed state.
+
+## Rationale
+
+Granda Mento is not intended to undermine the market or provide a way for large players to "dump" CELO for cUSD. It's also not intended to be an exceptionally elegant or complicated solution-- instead, this is looking to be something that is simple, straightforward, and works alongside the existing implementation of Mento. Any larger-scale Mento design modifications to facilitate large stable token mints are out of scope, and should be considered for a medium/long term design change to the stability protocol. The proposed implementation attempts to navigate the tradeoffs of complexity, centralization, economic safety, and ease of use for all actors in the Celo ecosystem.
```

Comment:
Do we want to create Granda Mento for both cUSD and cEUR? If yes, a generalization from cUSD to e.g. cXXX might be clarifying and increase consistency (As already used elsewhere here).
Another question that comes to my mind: What will happen for new stable tokens? Will they directly get a Granda Mento? @albertclabs